hazelcast/src/main/java/com/hazelcast/config/AbstractXmlConfigHelper.java

```diff
@@ -194,6 +195,10 @@ protected void schemaValidation(Document doc)
         } catch (Exception e) {
             throw new InvalidConfigurationException(e.getMessage());
         }
+        for (StreamSource source : schemas) {
+            closeResource(source.getInputStream());
```

Comment:
this should be in finally block too. I appreciate the method will look funny, but it's probably because it's doing too much.